aiida_quantumespresso/workflows/pdos.py

```diff
@@ -340,6 +340,7 @@ def get_builder_from_protocol(
         scf.pop('clean_workdir', None)
         nscf = PwBaseWorkChain.get_builder_from_protocol(*args, overrides=inputs.get('nscf', None), **kwargs)
         nscf['pw'].pop('structure', None)
+        nscf['pw']['parameters']['SYSTEM']['occupations'] = 'tetrahedra'
```

Comment:
what if the user explicitly specified a value for the `occupations` in the `nscf` step in the `overrides`? Even though this may be "incorrect", didn't we agree that the `overrides` should have the final say?

Reply:
Fair point! I suppose we could _only_ set `occupations` to `tetrahedra` if the setting is not in the `overrides`, and let the `validate_nscf` validation fail in case it's not set appropriately. 

Note that the only reason we have to do this here is because the `overrides` _don't_ have the final say in the `PwBaseWorkChain`, as `SYSTEM.occupations` gets adjusted by the `ElectronicType` _after_ the overrides are already merged with the protocol inputs.

Reply:
So I've tried to address this by specifically looking at the `overrides` in the `PdosWorkChain.get_builder_from_protocol()` method, but it all feels a little hacky to me. Now I'm specifically adding extra code to preserve an override that will simply cause the work chain to except anyways.

The problem lies with the fact that the `overrides` are not considered to be absolute in the lower level `PwBaseWorkChain`, which is why I need to add this extra line in the first place. But then the overrides are also not considered to be absolute when setting `RelaxType` in the `PwRelaxWorkChain`. So I think this is just a symptom of a larger problem that we discuss in https://github.com/aiidateam/aiida-quantumespresso/issues/745.

Since this PR is blocking https://github.com/aiidateam/aiida-quantumespresso/pull/747 and an improved solution for https://github.com/aiidateam/aiida-quantumespresso/pull/714, I'd be inclined to accept the fact that the incorrect overrides are ignored for now, until we come up with a solution for all the work chains.

Reply:
@sphuber I've bit the bullet and adapted the `PwBaseWorkChain.get_builder_from_protocol()` method so it follow the principle we decided on in #745, i.e. that the `overrides` are absolute. This avoids having to set the `SYSTEM.occupations` to `tetrahedra` in the `PdosWorkChain.get_builder_from_protocol()` method, since now the "protocol overrides" specified in the YAML file of the `PdosWorkChain` protocol is now applied correctly even if the `ElectronicType` is set to `Insulator`.

Additionally, this allowed me to remove the logic we added in #731, since now by default _all_ parameters provided in the overrides are recursively merged so they override process inputs generated by the protocol inputs. This means that the `overrides` _do_ take precedence over any supplied `initial_magnetic_moments` (which we can consider a protocol input). Note the change in the corresponding test.